group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1983,8 +1984,16 @@ private CoordinatorResult<StreamsGroupHeartbeatResult, CoordinatorRecord> stream
         StreamsGroupHeartbeatResponseData response = new StreamsGroupHeartbeatResponseData()
             .setMemberId(updatedMember.memberId())
             .setMemberEpoch(updatedMember.memberEpoch())
-            .setHeartbeatIntervalMs(streamsGroupHeartbeatIntervalMs(groupId))
-            .setPartitionsByUserEndpoint(maybeBuildEndpointToPartitions(group));
+            .setHeartbeatIntervalMs(streamsGroupHeartbeatIntervalMs(groupId));
+
+        if (group.endpointInformationEpoch() > memberEndpointEpochInRequest) {
```

Comment:
This is the fix, the group endpoint epoch is greater than the member's endpoint epoch indicating there's been a change and an update endpoint metadata is required.

Reply:
For the case where we lose the group's endpoint information during fail-over, shouldn't we also send the information, just in case?

Meaning that the condition should be 
```group.endpointInformationEpoch() != memberEndpointEpochInRequest```